pysat/model_utils.py

```diff
@@ -610,7 +610,8 @@ def extract_modelled_observations(inst=None, model=None, inst_name=[],
     # Update the instrument object and attach units to the metadata
     for mdat in interp_data.keys():
         attr_name = mdat.split("{:s}_".format(model_label))[-1]
-        inst.meta.data.units[mdat] = model.data_vars[attr_name].units
+        inst.meta.__setitem__(mdat, {inst.meta.units_label:
+                                     model.data_vars[attr_name].units})
```

Comment:
I'm not used to seeing setitem called this way.  Shouldn't it be hidden?

Reply:
This is the only way I've found to update the metadata.

Reply:
Do you have a code snippet I could run to test this?

Reply:
```
import pysat
testInst = pysat.Instrument('pysat', 'testing', '', clean_level='clean', update_files=True)
testInst.load(2009,1)
testInst.meta.__setitem__('dummy5', {testInst.meta.units_label: "Hi"})
print testInst.meta['dummy5']
```

Reply:
Oddly enough, the line I deleted works when I use it in ipython.  However, it fails in the routine.

Reply:
OK, I think the bug is generated because you'r setting units directly rather than creating a new key in `meta.data`.

A functionally equivalent version of line 613 would be:
```
inst.meta[mdat] =  {inst.meta.units_label: model.data_vars[attr_name].units}
```

This produces the same output using the test snippet above in 3.6.

Reply:
That hasn't worked before, but I can try it again later this week.

Reply:
What Jeff posted is how I remember it should work.

Reply:
You can also use inst.units_label, all the labels are available from both Meta and the instrument.

ivm.meta['lst'] = {ivm.units_label : 'bloop'}

ivm.meta['lst']
Out[17]: 
units                          bloop
long_name           Local Solar Time
notes        Local Solar Time at s/c
desc                Local Solar Time
label               Local Solar Time
axis          ICON_ANCILLARY_IVM_LST
scale                         linear
value_min                          0
value_max                         24
fill                          -999.9
children                        None
Name: lst, dtype: object

Reply:
Works in regular python as well

Reply:
OK FINE.  ;-P . Updated as suggested.  Also made some (more) PEP8 changes.